File: src/codes/Users_Manual/link/ethercrab_0.cpp
//~#include <iostream>
#include <autd3_link_ethercrab.hpp>

//~int main() {
//~using namespace autd3;
link::EtherCrab(
    [](const uint16_t idx, const link::Status status) {
      std::cout << "Device[" << idx << "]: " << status << std::endl;
    },
    link::EtherCrabOption{
        .ifname = std::nullopt,
        .state_check_period = std::chrono::milliseconds(100),
        .sync0_period = std::chrono::milliseconds(2),
        .sync_tolerance = std::chrono::microseconds(1),
        .sync_timeout = std::chrono::seconds(10),
    });
//~return 0; }

File: src/codes/Users_Manual/link/ethercrab_0.rs
use autd3_link_ethercrab::{EtherCrab, EtherCrabOption};
# use std::time::Duration;

# fn main() {
# let _ = 
EtherCrab::new(
    |idx, status| {
        eprintln!("Device[{}]: {}", idx, status);
    },
    EtherCrabOption {
        ifname: None,
        state_check_period: Duration::from_millis(100),
        sync0_period: Duration::from_millis(2),
        sync_tolerance: Duration::from_micros(1),
        sync_timeout: Duration::from_secs(10),
    },
);
# }